fcm.go

```diff
@@ -118,7 +120,64 @@ func (fcm *FCM) sendFirebaseRequest(payload Message) (*Response, error) {
 	}
 
 	// create request
-	req, err := http.NewRequest("POST", fcmEndpoint, bytes.NewBuffer(data))
+	req, err := http.NewRequest("POST", fcmSendEndpoint, bytes.NewBuffer(data))
+	if err != nil {
+		return nil, err
+	}
+
+	// add headers
+	req.Header.Set("Authorization", fmt.Sprintf("key=%s", fcm.app.apiKey))
+	req.Header.Set("Content-Type", "application/json")
+
+	// TODO use fcm.app.client if possible (not working actually)
+	client := &http.Client{}
+
+	// execute request
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+
+		switch resp.StatusCode {
+		case http.StatusBadRequest:
+			return nil, fmt.Errorf("StatusCode=%d, Desc=%s", resp.StatusCode, ErrInvalidParameters.Error())
+
+		case http.StatusUnauthorized:
+			return nil, fmt.Errorf("StatusCode=%d, Desc=%s", resp.StatusCode, ErrAuthentication.Error())
+
+		case http.StatusInternalServerError:
+		default:
+			return nil, fmt.Errorf("StatusCode=%d, Desc=%s", resp.StatusCode, ErrInternalServerError.Error())
+		}
+	}
+
+	// build response
+	response := new(Response)
+	if err := json.NewDecoder(resp.Body).Decode(response); err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
+
+func (fcm *FCM) sendFirebaseTopicRequest(endpoint string, payload Topic) (*Response, error) {
```

Comment:
sendFirebaseRequest, sendFirebaseTopicRequest, SubscribeDeviceToTopic, etc. should accept context then add context to request using
`req = req.WithContext(ctx)`